modules/account-lib/src/coin/hbar/transferBuilder.ts

```diff
@@ -10,66 +9,49 @@ import { isValidAddress, isValidAmount, stringifyAccountId } from './utils';
   protected async buildImplementation(): Promise<Transaction> {
-    this._txBodyData.transfers = this.buildTransferData();
+    this._cryptoTransferBuilder
+      .addSender(this._source.address, this._amount.toString())
+      .addRecipient(this._toAddress, this._amount.toString());
+    this._sdkTransactionBuilder = this._cryptoTransferBuilder;
+    this.transaction = await super.buildImplementation();
     this.transaction.setTransactionType(TransactionType.Send);
-    return await super.buildImplementation();
-  }
-
-  private buildTransferData(): proto.ITransferList {
-    return {
-      accountAmounts: [
-        { accountID: this.buildAccountData(this._source.address), amount: Long.fromString(this._amount).negate() }, // sender
-        { accountID: this.buildAccountData(this._toAddress), amount: Long.fromString(this._amount) }, // recipient
-      ],
-    };
-  }
-
-  private buildAccountData(address: string): proto.AccountID {
-    const accountData = new AccountId(address);
-    return new proto.AccountID({
-      accountNum: accountData.account,
-      realmNum: accountData.realm,
-      shardNum: accountData.shard,
-    });
-  }
-
-  /** @inheritdoc */
-  initBuilder(tx: Transaction): void {
-    super.initBuilder(tx);
-    this.transaction.setTransactionType(TransactionType.Send);
-    const transferData = tx.txBody.cryptoTransfer;
-    if (transferData && transferData.transfers && transferData.transfers.accountAmounts) {
-      this.initTransfers(transferData.transfers.accountAmounts);
-    }
+    return this.transaction;
   }
 
   /**
    * Initialize the transfer specific data, getting the recipient account
    * represented by the element with a positive amount on the transfer element.
    * The negative amount represents the source account so it's ignored.
    *
-   * @param {proto.IAccountAmount[]} transfers array of objects which contains accountID and transferred amount
+   * @param {Transaction} tx - the transaction data
    */
-  protected initTransfers(transfers: proto.IAccountAmount[]): void {
-    transfers.forEach(transferData => {
-      const amount = Long.fromValue(transferData.amount!);
-      if (amount.isPositive()) {
-        this.to(stringifyAccountId(transferData.accountID!));
-        this.amount(amount.toString());
-      }
-    });
+  initBuilder(tx: Transaction): void {
+    super.initBuilder(tx);
+    this.transaction.setTransactionType(TransactionType.Send);
+    const transferData = tx.txBody().getCryptotransfer();
+    if (transferData && transferData.getTransfers() && transferData.getTransfers()!.getAccountamountsList()) {
+      transferData
+        .getTransfers()!
+        .getAccountamountsList()
+        .forEach(transferData => {
+          const amount = Long.fromValue(transferData.getAmount());
+          if (amount.isPositive()) {
+            this.to(stringifyAccountId(transferData.getAccountid()!.toObject()));
+            this.amount(amount.toString());
+          }

```

Comment:
same as here - if they are non-positive, we should throw or do something

Reply:
We are taking the same approach here, checking that the account is different from the source account.